cms/datavis/blocks/base.py

```diff
@@ -246,6 +299,22 @@ def get_download_config(self, value: "StructValue") -> dict[str, Any]:
             ],
         }
 
+    def clean(self, value: "StructValue") -> "StructValue":
+        result = super().clean(value)
+        aspect_ratio_keys = [self.desktop_aspect_ratio, self.mobile_aspect_ratio]
+
+        options_errors = {}
+        if self.get_highcharts_chart_type(result) == "bar":
+            for i, option in enumerate(result["options"]):
+                if option.block_type in aspect_ratio_keys:
+                    options_errors[i] = ValidationError("Bar charts do not support aspect ratio options.")
+
+        if options_errors:
+            raise blocks.StructBlockValidationError(
+                block_errors={"options": blocks.StreamBlockValidationError(block_errors=options_errors)}
+            )
+        return result
```

Comment:
A minor suggestion to follow the "return early" pattern and reduce nesting:
```suggestion
    def clean(self, value: "StructValue") -> "StructValue":
        result = super().clean(value)
        
        if self.get_highcharts_chart_type(result) != "bar":
            return result

        options_errors = {}
        aspect_ratio_keys = [self.desktop_aspect_ratio, self.mobile_aspect_ratio]
        for i, option in enumerate(result["options"]):
            if option.block_type in aspect_ratio_keys:
                options_errors[i] = ValidationError("Bar charts do not support aspect ratio options.")

        if options_errors:
            raise blocks.StructBlockValidationError(
                block_errors={"options": blocks.StreamBlockValidationError(block_errors=options_errors)}
            )
        return result
```

Reply:
I expect in future there will be more validation here. There's an open question on the tick intervals ticket about whether the team wants us to validate that any axis maxima and minima are a multiple of the tick interval, for example. I appreciate the suggestion, but perhaps this is something to look at when we do code review for the charts feature as a whole.